src/main/java/SentinelRefactoring.java

```diff
@@ -0,0 +1,401 @@
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.List;
+
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.Assignment;
+import org.eclipse.jdt.core.dom.Block;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.ExpressionStatement;
+import org.eclipse.jdt.core.dom.IfStatement;
+import org.eclipse.jdt.core.dom.InfixExpression;
+import org.eclipse.jdt.core.dom.NullLiteral;
+import org.eclipse.jdt.core.dom.NumberLiteral;
+import org.eclipse.jdt.core.dom.PrefixExpression;
+import org.eclipse.jdt.core.dom.SimpleName;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.VariableDeclarationFragment;
+import org.eclipse.jdt.core.dom.VariableDeclarationStatement;
+
+/**
+ * This class represents a refactoring in which integer variables whose values
+ * represent the nullness of another variables are refactored into explicit null
+ * checks
+ */
+public class SentinelRefactoring extends Refactoring {
+	public static final String NAME = "SentinelRefactoring";
+	/**
+	 * List of all sentinels found during traversal of the AST
+	 */
```

Comment:
This Javadoc comment should explain the format of the key, which is a raw String.

Reply:
e.g., is it a fully-qualified name? Something else? And, how do you know that the keys will be unique? (A proof sketch is a good idea: we need to be sure that there won't be collisions due to variables with the same name (shadowing), variables declared on the same line, etc.)

Reply:
This comment hasn't been addressed @AtomicPhoenix